extensions/liquibase/deployment/src/main/java/io/quarkus/liquibase/LiquibaseProcessor.java

```diff
@@ -0,0 +1,319 @@
+                liquibase.datatype.LiquibaseDataType.class,
+                liquibase.executor.Executor.class,
+                liquibase.lockservice.LockService.class,
+                liquibase.sqlgenerator.SqlGenerator.class)
+                .forEach(t -> addService(reflective, generatedResourceProducer, resourceProducer, t));
+    }
+
+    /**
+     * Build the static init phase of the liquibase extension
+     * 
+     * @param additionalBeanProducer the additional bean producer
+     * @param featureProducer the feature producer
+     * @param resourceProducer the resource producer
+     * @param containerListenerProducer the container listener producer
+     * @param recorder the liquibase recorder
+     * @param dataSourceInitializedBuildItem Added this dependency to be sure that Agroal is initialized first
+     * @param generatedBeanBuildItem the generated bean build item
+     */
+    @Record(STATIC_INIT)
+    @BuildStep(loadsApplicationClasses = true)
+    void build(BuildProducer<AdditionalBeanBuildItem> additionalBeanProducer,
+            BuildProducer<FeatureBuildItem> featureProducer,
+            BuildProducer<NativeImageResourceBuildItem> resourceProducer,
+            BuildProducer<BeanContainerListenerBuildItem> containerListenerProducer,
+            LiquibaseRecorder recorder,
+            DataSourceInitializedBuildItem dataSourceInitializedBuildItem,
+            BuildProducer<GeneratedBeanBuildItem> generatedBeanBuildItem,
+            BuildProducer<NativeImageResourceBundleBuildItem> resourceBundle) {
+
+        featureProducer.produce(new FeatureBuildItem(FeatureBuildItem.LIQUIBASE));
+
+        AdditionalBeanBuildItem unremovableProducer = AdditionalBeanBuildItem.unremovableOf(LiquibaseProducer.class);
+        additionalBeanProducer.produce(unremovableProducer);
+
+        Collection<String> dataSourceNames = DataSourceInitializedBuildItem
+                .dataSourceNamesOf(dataSourceInitializedBuildItem);
+        new LiquibaseDatasourceBeanGenerator(dataSourceNames, generatedBeanBuildItem).createLiquibaseProducerBean();
+
+        containerListenerProducer.produce(
+                new BeanContainerListenerBuildItem(recorder.setLiquibaseBuildConfig(liquibaseBuildConfig)));
+
+        resourceProducer.produce(
+                new NativeImageResourceBuildItem(getChangeLogs(dataSourceInitializedBuildItem).toArray(new String[0])));
+
+        // liquibase XSD
+        resourceProducer.produce(new NativeImageResourceBuildItem(
+                "www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.5.xsd",
+                "www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.6.xsd",
+                "www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.7.xsd",
+                "www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.8.xsd",
+                "www.liquibase.org/xml/ns/dbchangelog/dbchangelog-ext.xsd",
+                "www.liquibase.org/xml/ns/pro/liquibase-pro-3.8.xsd",
```

Comment:
Can you check if `NativeImageResourceBuildItem` accepts regexps? I would be more permissive here to avoid having to update that in the future.

Reply:
I tried but it doesn't work for me.
```
        resourceProducer.produce(new NativeImageResourceBuildItem(
                ".*properties$",
                "www.liquibase.org/xml/ns/.*$"));
```
The native image of the liquibase has this configuration
https://github.com/1000kit/liquibase/blob/9ebb46978344aab2f4a531e61ec6c7d20a656fbc/src/main/resources/META-INF/native-image/org.tkit.liquibase/liquibase/resource-config.json#L9

Reply:
@gsme please check the regex for the native resources build item:
https://github.com/andrejpetras/quarkus/commit/0c5eee023ec085ff77470d485f42a4cfe90b289c

Reply:
https://github.com/quarkusio/quarkus/issues/7033